packages/utils/core-utils/test/package-manager_spec.ts

```diff
@@ -0,0 +1,35 @@
+import { expect } from 'chai';
+
+import { getPackageManager } from '../src/package-manager';
+
+describe('get-package-manager', () => {
+  let userAgent: string | undefined;
+
+  beforeEach(() => {
+    userAgent = process.env.npm_config_user_agent;
+    delete process.env.npm_config_user_agent;
+  });
+
+  afterEach(() => {
+    if (!userAgent) {
+      delete process.env.npm_config_user_agent;
+    } else {
+      process.env.npm_config_user_agent = userAgent;
+    }
+  });
+
+  it('should return npm if npm_config_user_agent=npm', () => {
+    process.env.npm_config_user_agent = 'npm';
```

Comment:
nit: can we test realer user agent data (e.g. `npm/8.19.2 node/v18.12.1 darwin x64`)?